src/pages/topology/Servers.js

```diff
@@ -0,0 +1,353 @@
+// (c) Copyright 2017-2018 SUSE LLC
+/**
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+import React, { Component } from 'react';
+import { HashLink } from 'react-router-hash-link';
+import { translate } from '../../localization/localize.js';
+import '../../styles/deployer.less';
+import { getInternalModel, noHyphenWrap } from './TopologyUtils.js';
+import { ErrorBanner } from '../../components/Messages';
+import { LoadingMask } from '../../components/LoadingMask';
+import { byProperty, byEntry } from '../../utils/Sort.js';
+
+/*
+ * This class is a JavaScript implementation of the script
+ * ardana_configurationprocessor/plugins/builders/HTMLDiagram/ServerGroups.py and
+ * ardana_configurationprocessor/plugins/builders/HTMLDiagram/Servers.py
+ * in the config processor
+ */
+class Servers extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      model: undefined,
+      errorMessage: undefined,
+    };
+
+  }
+
+  componentDidMount() {
+
+    getInternalModel()
+      .then((yml) => {
+
+        // Force a re-render if the page is still shown (user may navigate away while waiting)
+        if (this.refs.server_groups)
```

Comment:
I'm not sure I understand this section, though it seems to work ok and its not technically new since it existed in the previous version of this file (which was renamed and git doesnt handle that well)